src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilder.java

```diff
@@ -0,0 +1,447 @@
+                }
+                return distributionKey;
+            default:
+                throw new UnsupportedOperationException("Distribution type not allowed for generator");
+        }
+        return weight;
+    }
+
+    private double getLoadWeight(Load load, SensitivityAnalysisInputData.DistributionType distributionType, Double distributionKey) {
+        double weight;
+        switch (distributionType) {
+            case PROPORTIONAL:
+            case PROPORTIONAL_MAXP: // simpler to use the same enum for generator and load
+                weight = load.getP0();
+                break;
+            case REGULAR:
+                weight = 1.;
+                break;
+            case VENTILATION:
+                if (distributionKey == null) {
+                    throw new PowsyblException("Distribution key required for VENTILATION distribution type !!");
+                }
+                return distributionKey;
+            default:
+                throw new UnsupportedOperationException("Distribution type not allowed for load");
+        }
+        return weight;
+    }
+
+    private List<SensitivityVariableSet> buildSensitivityVariableSets(List<IdentifiableType> variablesTypesAllowed,
+                                                                      List<SensitivityAnalysisInputData.Ident> variablesFiltersListIdents,
+                                                                      SensitivityAnalysisInputData.DistributionType distributionType,
+                                                                      Reporter reporter) {
+        List<SensitivityVariableSet> result = new ArrayList<>();
+
+        List<List<IdentifiableAttributes>> variablesFiltersLists = variablesFiltersListIdents.stream()
+            .map(variablesFilterIdent -> {
+                List<IdentifiableAttributes> list = filterService.getIdentifiablesFromFilter(variablesFilterIdent.getId(), context.getNetworkUuid(), context.getVariantId());
+                if (list.stream().allMatch(i -> variablesTypesAllowed.contains(i.getType()))) {
+                    return list;
+                } else {
+                    reporter.report(Report.builder()
+                        .withKey("badEquipmentType")
+                        .withDefaultMessage("Equipments type in filter with name=${name} should be ${expectedType} : filter is ignored")
+                        .withValue("name", variablesFilterIdent.getName())
+                        .withValue(EXPECTED_TYPE, variablesTypesAllowed.toString())
+                        .withSeverity(TypedValue.WARN_SEVERITY)
+                        .build());
+                    return Collections.<IdentifiableAttributes>emptyList();
+                }
+            })
+            .collect(Collectors.toList());
+
+        variablesFiltersLists.forEach(variablesList -> {
+            List<WeightedSensitivityVariable> variables = new ArrayList<>();
+            if (!variablesList.isEmpty()) {
```

Comment:
You may pre-filter out these empty variables list, or return on such empty cases.

Reply:
Done